File: python/towerOfHanoi.py
def towers(disc_count):
    tower_1 = [i for i in range(1, disc_count + 1)]
    tower_2 = []
    tower_3 = []

    def move_discs(disc_count, origin, stage, goal):
        if (disc_count == 1):
            disc = origin.pop(0)
            goal.insert(0, disc)

            return goal
        
        move_discs(disc_count - 1, origin, goal, stage)

        disc = origin.pop(0)
        goal.insert(0, disc)

        move_discs(disc_count - 1, stage, origin, goal)

        return goal
    return move_discs(disc_count, tower_1, tower_2, tower_3)

File: javascript/towerOfHanoi.js
const towers = (discCount) => {
    const tower1 = [...Array(discCount + 1).keys()].slice(1);
    const tower2 = [], tower3 = [];

    const moveDiscs = (discCount, origin, stage, goal) => {
        if (discCount === 1) {
            let disc = origin.shift();
            goal.unshift(disc);
        return goal;
    }

        moveDiscs(discCount - 1, origin, goal, stage);

        let disc = origin.shift();
        goal.unshift(disc);

        moveDiscs(discCount - 1, stage, origin, goal);
            return goal;
    }

    return moveDiscs(discCount, tower1, tower2, tower3);
}